### PR TITLE
Adds pull-image

### PR DESCRIPTION
**Pull request by Itxaka**
Basically a copy of luet's util unpack command

Signed-off-by: Itxaka <igarcia@suse.com>

### CODE REVIEW

File: cmd/pull-image.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
+/*
+Copyright © 2021 SUSE LLC
```
Comment (mudler) [inline cmd/pull-image.go]:
```suggestion
Copyright © 2022 SUSE LLC
```

**Review by mudler — APPROVED** [review]

File: cmd/pull-image.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+		auth := &types.AuthConfig{
+			Username:      user,
+			Password:      pass,
+			ServerAddress: server,
+			Auth:          authType,
+			IdentityToken: identity,
+			RegistryToken: registryToken,
+		}
+
+		info, err := docker.DownloadAndExtractDockerImage(util.DefaultContext, image, destination, auth, verify)
```
Comment (davidcassany) [inline cmd/pull-image.go]: I would use and extend the Luet interface instead, so we have a single point of connection between luet and elemental. 

https://github.com/rancher-sandbox/elemental/blob/30ddad2f81f189297b283037bdb4d6e9613e3a8f/pkg/types/v1/luet.go#L50-L60

We could use a map for extra configs, pass the auth struct as parameter, make it part of the constructor of NewLuet, etc.

In any case I'd try to keep luet wrapped into the interface.

**Review by davidcassany — COMMENTED** [review]
I am just wondering if we could eventually extend the interface of Luet and make use of it here.

File: cmd/pull-image.go (hunk 2 above)
Comment (Itxaka) [reply cmd/pull-image.go]: Uh, didn't even knew that we had that already, so yeah I will change to use the luet.unpack directly indeed.

Thanks for the pointer!

Comment (Itxaka) [reply cmd/pull-image.go]: There, actually is much nicer :D

File: cmd/root.go [hunk 3]
```diff
@@ -30,7 +31,10 @@ var rootCmd = &cobra.Command{
 // Execute adds all child commands to the root command and sets flags appropriately.
 // This is called by main.main(). It only needs to happen once to the rootCmd.
 func Execute() {
-	cobra.CheckErr(rootCmd.Execute())
+	err := rootCmd.Execute()
```
Comment (Itxaka) [inline cmd/root.go:34]: Otherwise, this prints the error to stdout, so we better pick the error and exit ourselves

**Review by davidcassany — APPROVED** [review]
LGTM

**Pull request merged**